core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -214,6 +220,69 @@ public void declareField(Parser<Value, Context> p, ParseField parseField, ValueT
         declareField((p, v, c) -> consumer.accept(v, parser.parse(p, c)), parseField, type);
     }
 
+    /**
+     * Declares a parser for fields of which names are not known upfront when creating the ObjectParser.
+     * As an example, in the aggregation output we have things like:
+     * <pre><code>
+     * "aggregations" : {
+     *   "genres" : {  <--- arbitrary aggregation name
+     *      "doc_count_error_upper_bound": 0,
+     *      "sum_other_doc_count": 0,
+     *      "buckets" : [
+     *          {
+     *              "key" : "jazz",
+     *              "doc_count" : 10
+     *              "total_number_of_ratings": { <--- arbitrary aggregation name
```

Comment:
I updated the example. The test should now cover a more complex case with inner objects, although I didn't use any special aggregation parsers in the test to keep them more simple.